### PR TITLE
Model submission for the CERBERUS ANYmal C platform sensor configurations 1 and 2 by CERBERUS

### PR DESCRIPTION
**Pull request by Samu24**
PR adds CERBERUS ANYmal C sensor configuration 1 & 2.

Sensor configuration 1 includes the basic setup for the ANYmal C robot.
Sensor configuration 2 additionally includes 12 breadcrumbs for deployment.

This PR depends on the controller update in https://github.com/leggedrobotics/cerberus_anymal_locomotion/pull/10.

Notes:
The thumbnail script did not work during the preparation. Therefore the thumbnail images are not added yet.

### CODE REVIEW

**Comment by Samu24** [comment]
fyi @marco-tranzatto

**Comment by Samu24** [comment]
PR https://github.com/leggedrobotics/subt_cerberus_anymal/pull/1 can be merged into this PR once the recompiled plugin for the dome release has been tested.

**Comment by Samu24** [comment]
@angelacmaio Dome plugin update is merged into this PR.

**Review by azeey — APPROVED** [review]
This looks good to me. I have tested it locally with the controller from https://github.com/leggedrobotics/cerberus_anymal_locomotion/pull/10. One thing I noticed was that the robot doesn't have a front facing light. Is that intentional?

**Comment by nkoenig** [comment]
@Samu24, can you address the comment by @azeey? Let me know if you have any questions or need assistance.

File: submitted_models/cerberus_anymal_c_sensor_config_1/model.sdf [hunk 1]
```diff
@@ -0,0 +1,1772 @@
...
+      <sensor name="camera_0" type="camera">
+        <always_on>1</always_on>
+        <update_rate>20</update_rate>
+        <pose frame="">-0.400 -0.050 0.229 -0.006 0.011 3.133</pose>
+        <camera name="camera_0">
+          <horizontal_fov>2.19911</horizontal_fov>
+          <image>
+            <width>720</width>
+            <height>540</height>
+            <format>L8</format>
```
[the remark this answers is not in the document]
Comment (Samu24) [reply submitted_models/cerberus_anymal_c_sensor_config_1/model.sdf:644]: @azeey, we can also switch to `R8G8B8` format. Regarding the monochrome and color cameras. We mention in `specifictions.md` in the section `Diversions from Physical Hardware of ANYmal` that all cameras are modeled as color cameras. I would say best is probably to switch all of them to `R8G8B8`. What do you think?

**Comment by Samu24** [comment]
@nkoenig, yes I will have a look.
@azeey, thanks for the review.

The robot should have a forward facing light. There is one in the `model.sdf`. I can check this again once I manage to overcome #728.

File: submitted_models/cerberus_anymal_c_sensor_config_1/model.sdf (hunk 1 above)
[the remark this answers is not in the document]
Comment (azeey) [reply submitted_models/cerberus_anymal_c_sensor_config_1/model.sdf:644]: Yes, that sounds good. You can simply remove `<format>` or comment it out in that case.

File: submitted_models/cerberus_anymal_c_sensor_config_1/model.sdf [hunk 2]
```diff
@@ -0,0 +1,1772 @@
...
+          </noise>
+        </lidar>
+        <pose frame="">-0.524 -0.000 -0.042 -2.576 0.245 1.206</pose>
+      </sensor>
+      <sensor name="camera_0" type="camera">
+        <always_on>1</always_on>
+        <update_rate>20</update_rate>
+        <pose frame="">-0.400 -0.050 0.229 -0.006 0.011 3.133</pose>
+        <camera name="camera_0">
+          <horizontal_fov>2.19911</horizontal_fov>
```
Comment (azeey) [inline submitted_models/cerberus_anymal_c_sensor_config_1/model.sdf:640]: @mjcarroll noticed that these cameras don't have lens properties (specified in the `<lens>` tag). Can you please add them?

**Review by azeey — CHANGES_REQUESTED** [review]

File: submitted_models/cerberus_anymal_c_sensor_config_1/model.sdf (hunk 2 above)
Comment (mjcarroll) [reply submitted_models/cerberus_anymal_c_sensor_config_1/model.sdf:640]: Using our standard formulae, I would recommend:

```
<intrinsics>
    <!-- fx = fy = width / ( 2 * tan (hfov / 2 ) ) -->
    <fx>183.43</fx>
    <fy>183.43</fy>
    <!-- cx = ( width + 1 ) / 2 -->
    <cx>360.5</cx>
    <!-- cy = ( height + 1 ) / 2 -->
    <cy>270.5</cy>
    <s>0</s>
</intrinsics>
```

**Comment by acschang** [comment]
> The robot should have a forward facing light. There is one in the `model.sdf`. I can check this again once I manage to overcome #728.

It seems the light labeled as "forward" actually faces the "rear" with respect to the vehicle spawn and orientation of a positive X velocity request.

**Comment by Samu24** [comment]
@acschang, thanks for this comment. I think I see the confusion now. In reality we are using the robot to walk in negative x direction all the time as the cameras and lights are in the rear of the robot. That means "forward" on the real robot is rather "backward"

Given that, would it make sense to introduce a dummy frame that connects the real base to the sim base?

File: submitted_models/cerberus_anymal_c_sensor_config_1/specifications.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+rostopic pub /anymal_c/cmd_vel geometry_msgs/Twist "linear:
+  x: 0.5
+  y: 0.0
+  z: 0.0
+angular:
+  x: 0.0
+  y: 0.0
+  z: 0.0"
+```
+
```
Comment (acschang) [inline submitted_models/cerberus_anymal_c_sensor_config_1/specifications.md:37]: Please add a note mentioning that CERBERUS commands _negative X velocity to move forward_ on the ANYmal due to the positioning of the mounted sensors.

**Review by acschang — CHANGES_REQUESTED** [review]
Please resolve the comment regarding vehicle orientation in the `specifications.md` file.

**Comment by acschang** [comment]
> @acschang, thanks for this comment. I think I see the confusion now. In reality we are using the robot to walk in negative x direction all the time as the cameras and lights are in the rear of the robot. That means "forward" on the real robot is rather "backward"
> 
> Given that, would it make sense to introduce a dummy frame that connects the real base to the sim base? I.e. that the robot is rotated 180 degrees and a positive command makes the robot walk in the direction of the cameras and the lights.

Thank you for the clarification.  A simple note on your convention for operation in the **Usage Instructions** section of the `specifications.md` file will be sufficient.

File: submitted_models/cerberus_anymal_c_sensor_config_1/model.sdf (hunk 2 above)
Comment (acschang) [reply submitted_models/cerberus_anymal_c_sensor_config_1/model.sdf:640]: @Samu24 can you please follow mjcarroll's suggested lens intrinsic parameters?  I believe this is the last outstanding issue prior to acceptance.

**Comment by nkoenig** [comment]
The bounding box for this model is
```
Min[-0.599876 -0.316324 -0.503438] Max[0.599877 0.315419 0.284477]
```

File: submitted_models/cerberus_anymal_c_sensor_config_1/model.sdf (hunk 2 above)
Comment (Samu24) [reply submitted_models/cerberus_anymal_c_sensor_config_1/model.sdf:640]: Yes I can take care of that.

Comment (Samu24) [reply submitted_models/cerberus_anymal_c_sensor_config_1/model.sdf:640]: @acschang @azeey
Lens properties have been added.

**Review by acschang — APPROVED** [review]

**Pull request merged**